bentoml/deployment/sagemaker/__init__.py

```diff
@@ -343,27 +358,213 @@ def delete(self):
+        )
+        logger.info(
+            "Creating Sagemaker endpoint %s configuration", endpoint_config_name
+        )
+        create_endpoint_config_response = sagemaker_client.create_endpoint_config(
+            EndpointConfigName=endpoint_config_name,
+            ProductionVariants=production_variants,
+        )
+        logger.info(
+            "AWS create endpoint config response: %s", create_endpoint_config_response
+        )
+
+        endpoint_name = generate_aws_compatible_string(
+            deployment_pb.namespace + '-' + deployment_spec.bento_name
+        )
+        logger.info("Creating sagemaker endpoint %s", endpoint_name)
+        create_endpoint_response = sagemaker_client.create_endpoint(
+            EndpointName=endpoint_name, EndpointConfigName=endpoint_config_name
+        )
+        logger.info("AWS create endpoint response: %s", create_endpoint_response)
+
+        res_deployment_pb = Deployment(state=DeploymentState())
         res_deployment_pb.CopyFrom(deployment_pb)
-        # res_deployment_pb.state = ...
+
+        # We are going to wait 10 mins for AWS to create sagemaker
+        start_time = time.time()
+        while (time.time() - start_time) < 600:
+            state = self.describe(res_deployment_pb, repo).state
+            if state.state == DeploymentState.PENDING:
+                time.sleep(10)
+                continue
+            else:
+                break
+        res_deployment_pb.state.CopyFrom(state)
+
         return ApplyDeploymentResponse(status=Status.OK(), deployment=res_deployment_pb)
 
-    def delete(self, deployment_pb, repo):
-        # delete deployment
+    def delete(self, deployment_pb, repo=None):
+        deployment_spec = deployment_pb.spec
+        sagemaker_config = deployment_spec.sagemaker_operator_config
+        if sagemaker_config is None:
+            raise BentoMLDeploymentException('Sagemaker configuration is missing.')
+        sagemaker_client = boto3.client('sagemaker', sagemaker_config.region)
+
+        endpoint_name = generate_aws_compatible_string(
+            deployment_pb.namespace + '-' + deployment_spec.bento_name
+        )
+        delete_endpoint_response = sagemaker_client.delete_endpoint(
+            EndpointName=endpoint_name
+        )
+        logger.info("AWS delete endpoint response: %s", delete_endpoint_response)
+        if delete_endpoint_response["ResponseMetadata"]["HTTPStatusCode"] == 200:
+            # We will also try to delete both model and endpoint configuration for user.
+            # Since they are not critical, even they failed, we will still count delete
+            # deployment a success action
+            model_name = create_sagemaker_model_name(
+                deployment_spec.bento_name, deployment_spec.bento_version
+            )
+            delete_model_response = sagemaker_client.delete_model(ModelName=model_name)
+            logger.info("AWS delete model response: %s", delete_model_response)
```

Comment:
I feel it's a bit noisy for users to see all the intermediate response. In the case of failure, it makes sense to show the error which is handled below, otherwise, maybe a simple sentence like _Modle is successfully deleted_ is sufficient. (Or maybe we log the response in Debug level?)

Reply:
I think you are right on too much info.  This logs are left over from previous to work with CLI.  We should clean these up, might be another pr.